app/qt_controller.cpp

```diff
@@ -0,0 +1,34 @@
+/// \file
+/// Maintainer: Luzian Hug
+///
+///
+
+#include "qt_controller.h"
+
+#include <QApplication>
+#include <QPushButton>
+
+namespace MouseTrack {
+
+int QtController::main(int argc, char *argv[]){
+
+    // set up application
+    QApplication app (argc, argv);
+
+    // show a button in a window
+    QPushButton button ("Hello world !");
+    button.setToolTip("A tooltip");
+    button.show();
+    // start pipeline
+    pipeline().start();
```

Comment:
@luhug : the entire method is just a draft, feel free to move stuff around as it is convenient for you.

Reply:
What's controller.cpp and what is qt_controller.cpp?

Reply:
`Controller` defines an abstract class, controller.cpp implements the interface for the pipeline access. Each controller is responsible to connect the ui with the model code (the part that does the actual work). I've decided to put the pipeline as a member into the controller class, since every controller will interact with a pipeline, at least in our case. QtController is a child of controller. It knows how to create a qt gui and interact with the user. I've also written a cliController that allows us to perform integration tests (comes with a later push).